tasm-test/src/DatabaseTest.cc

```diff
@@ -0,0 +1,140 @@
+
+        std::string video = "video";
+        unsigned int id = 0;
+
+        std::vector<unsigned int> heights{1, 2, 3};
+        std::vector<unsigned int> widths{1, 2, 3, 4};
+        unsigned int numColumns = widths.size();
+        unsigned int numRows = heights.size();
+        std::shared_ptr<TileLayout> tileLayout = std::make_shared<TileLayout>(numColumns, numRows, widths, heights);
+
+        layoutDatabase->addTileLayout(video, id, 40, 50, tileLayout);
+
+        std::shared_ptr<TileLayout> selectedLayout = layoutDatabase->tileLayoutForId(video, id);
+        assert(selectedLayout->numberOfColumns() == numColumns);
+        assert(selectedLayout->numberOfRows() == numRows);
+        assert(selectedLayout->widthsOfColumns() == widths);
+        assert(selectedLayout->heightsOfRows() == heights);
+        assert(*selectedLayout == *tileLayout);
+    }
+
+    TEST_F(DatabaseTestFixture, testSelectLayoutIds) {
+        std::shared_ptr<LayoutDatabase> layoutDatabase = LayoutDatabase::instance();
+
+        std::string video = "video";
+
+        std::vector<unsigned int> heights{1, 2, 3};
+        std::vector<unsigned int> widths{1, 2, 3, 4};
+        unsigned int numColumns = widths.size();
+        unsigned int numRows = heights.size();
+        std::shared_ptr<TileLayout> tileLayout = std::make_shared<TileLayout>(numColumns, numRows, widths, heights);
+
+        std::vector<int> layoutIds{0, 2, 4, 7};
+
+        for (auto &id : layoutIds) {
+            layoutDatabase->addTileLayout(video, id, id, id + 10, tileLayout);
+        }
+
+        // add an id that won't be selected, since it doesn't include frame 10
+        layoutDatabase->addTileLayout(video, 10, 11, 12, tileLayout);
+
+        std::vector<int> foundLayoutIds = layoutDatabase->tileLayoutIdsForFrame(video, 10);
+        std::sort(layoutIds.begin(), layoutIds.end());
+        std::sort(foundLayoutIds.begin(), foundLayoutIds.end());
+        assert(layoutIds == foundLayoutIds);
+    }
+
+    TEST_F(DatabaseTestFixture, testTileLocation) {
+        std::shared_ptr<LayoutDatabase> layoutDatabase = LayoutDatabase::instance();
+
+        std::string name = "name";
+
+        int tileNumber = 4;
+        unsigned int id = 2;
+        unsigned int firstFrame = 40;
+        unsigned int lastFrame = 59;
+        std::vector<unsigned int> heights{1, 2};
+        std::vector<unsigned int> widths{1, 2};
+        unsigned int numColumns = 2;
+        unsigned int numRows = 2;
+        std::shared_ptr<TileLayout> tileLayout = std::make_shared<TileLayout>(numColumns, numRows, widths, heights);
+
+        layoutDatabase->addTileLayout(name, id, firstFrame, lastFrame, tileLayout);
+
+        auto entry = std::make_shared<TiledEntry>(name);
+
+        std::experimental::filesystem::path expectedLocation = "";
+
+        std::experimental::filesystem::path tileLocation = layoutDatabase->locationOfTileForId(entry, tileNumber, id);
```

Comment:
What is this test actually checking? It doesn't look like expectedLocation and tileLocation are ever used.